src/model/net.py

```diff
@@ -1,4 +1,127 @@
-"""
-    This script was made by Nick at 19/07/20.
-    To implement code of your network using operation from ops.py.
-"""
+import torch
+import torch.nn as nn
+
+from omegaconf import DictConfig
+from torchsummary import summary as torch_summary
+
+from src.nn.binarized_linear import BinaryLinear
+from src.nn.binarized_conv2d import BinarizedConv2d
+
+class BinaryLinear(nn.Module):
+    def __init__(self, model_config: DictConfig):
+        super(BinaryLinear, self).__init__()        
+        mode = str(model_config.params.mode)
+        classes = int(model_config.params.classes)
+        dropout_ratio = float(model_config.params.dropout_ratio)        
+        self.width = int(model_config.params.width)
+        self.height = int(model_config.params.height)
+        self.channels = int(model_config.params.channels)
+
+        self.fc1 = BinaryLinear(width * height * channels, 1024, bias=False, mode="Stocastic")
+        self.batch1 = nn.BatchNorm1d(1024)
+        self.fc2 = BinaryLinear(1024, 1024, bias=False, mode=mode)
+        self.batch2 = nn.BatchNorm1d(1024)
+        self.fc3 = BinaryLinear(1024, 1024, bias=False, mode=mode)
+        self.batch3 = nn.BatchNorm1d(1024)
+        self.fc4 = BinaryLinear(1024, classes, bias=False, mode=mode)
+        self.relu = nn.ReLU()
```

Comment:
지금 보면 init에 반복되는 building block 이 등장하는데요, 이런 경우 하나의 반복 최소 단위를 함수로 만들면 가독성도 좋아지고, 코드가 깔끔해져요. 🤗 

```suggestion
    def linear_block(dim_in, dim_out, bias=False, mode="Stochastic"):
        return nn.Sequential(
            BinaryLinear(dim_in, dim_out, bias=bias, mode=mode),
            nn.BatchNorm1d(dim_out),
            nn.ReLU()
        )

        self.linear_block1 = linear_block(width * height * channels, 1024)
        self.linear_block2 = linear_block(1024, 1024)
        self.linear_block3 = linear_block(1024, 1024)
        self.linear = BinaryLinear(1024, classes)
```